cmd/wstail/matcher.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+func buildMatcher(filter filter) string {
+	if filter.freeText != "" {
+		return regexp.QuoteMeta(strings.TrimSpace(filter.freeText))
+	}
+	containerMatcher := "(" + strings.Join(filter.containers, "|") + ")"
+	podNamespace := match.placeholder
+	var pods []string
+	if len(filter.pods) > 0 {
+		pods = filter.pods
+	} else {
+		pods = []string{match.placeholder}
+	}
+
+	var podMatchers []string
+	for _, podName := range pods {
+		podMatchers = append(podMatchers, fmt.Sprintf(`%v_%v_%v`, podName, podNamespace, containerMatcher))
+	}
+	return "^" + prefix + match.whitespace + "(" + strings.Join(podMatchers, "|") + ")"
+}
+
+// Macthers
+var match = struct {
```

Comment:
i kind of wanted the scope with `match` but I think you're right - scratch that, constants will do